### PR TITLE
Refactor test cases of AnnotateRoutes as for Rake versions

### PR DESCRIPTION
**Pull request by nard-tech**
Refactor test cases for entire rewrite in the future.

### CODE REVIEW

**Comment by nard-tech** [comment]
CI for 2.4.x is failed due to recent update of [seattlerb/sexp_processor](https://github.com/seattlerb/sexp_processor).

I made a PR of `sexp_processor` to fix this CI failure, so please wait for a while.

**Comment by nard-tech** [comment]
[seattlerb/sexp_processor](https://github.com/seattlerb/sexp_processor) was fixed and CI have passed.

**Comment by nard-tech** [comment]
I fixed conflict of `.rubocop_todo.yml`

File: spec/lib/annotate/annotate_routes_spec.rb [hunk 1]
```diff
@@ -272,14 +272,15 @@
...
-        expect(AnnotateRoutes).to receive(:`).with('rake routes').and_return(rake_routes_result)
-        expect(File).to receive(:open).with(ROUTE_FILE, 'wb').and_yield(mock_file)
-        expect(AnnotateRoutes).to receive(:puts).with(MESSAGE_ANNOTATED)
-      end
+    before :each do
+      expect(File).to receive(:exist?).with(ROUTE_FILE).and_return(true)
+      expect(File).to receive(:open).with(ROUTE_FILE, 'wb').and_yield(mock_file)
+      expect(File).to receive(:read).with(ROUTE_FILE).and_return(route_file_content)
+
+      expect(AnnotateRoutes).to receive(:`).with('rake routes').and_return(rake_routes_result)
```
Comment (drwl) [inline spec/lib/annotate/annotate_routes_spec.rb]: This is interesting 🤔. I had no idea that when you do 
```
`rake routes`
```

It sends the message with method name `. Very neat.

File: spec/lib/annotate/annotate_routes_spec.rb [hunk 2]
```diff
@@ -333,21 +335,15 @@
         end
 
         it 'annotates without an empty line' do
-          expect(File).to receive(:read).with(ROUTE_FILE).and_return(route_file_content)
           expect(mock_file).to receive(:puts).with(expected_result)
+          expect(AnnotateRoutes).to receive(:puts).with(MESSAGE_ANNOTATED)
```
Comment (drwl) [inline spec/lib/annotate/annotate_routes_spec.rb]: Same with `puts`, I didn't know that the object receiving the message is whatever object that it's being called in.

**Pull request merged**